pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -71,6 +71,71 @@ func NewStaticResourceController(
 	return c
 }
 
+func (c *StaticResourceController) AddKubeInformers(kubeInformersByNamespace v1helpers.KubeInformersForNamespaces) *StaticResourceController {
+	ret := c
+	for _, file := range c.files {
+		objBytes, err := c.manifests(file)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("missing %q: %v", file, err))
+			continue
+		}
+		requiredObj, _, err := genericCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("cannot decode %q: %v", file, err))
+			continue
+		}
+		metadata, err := meta.Accessor(requiredObj)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("cannot get metadata %q: %v", file, err))
+			continue
+		}
+
+		// find the right subset of informers.  Interestingly, cluster scoped resources require cluster scoped informers
+		var informer informers.SharedInformerFactory
+		if _, ok := requiredObj.(*corev1.Namespace); !ok {
+			informer := kubeInformersByNamespace.InformersFor(metadata.GetName())
+			if informer == nil {
+				utilruntime.HandleError(fmt.Errorf("missing informer for namespace %q; no dynamic wiring added, time-based only.", metadata.GetName()))
+				continue
+			}
+		} else {
+			informer := kubeInformersByNamespace.InformersFor(metadata.GetNamespace())
+			if informer == nil {
+				utilruntime.HandleError(fmt.Errorf("missing informer for namespace %q; no dynamic wiring added, time-based only.", metadata.GetNamespace()))
+				continue
+			}
+		}
+
+		// iterate through the resources we know that are related to kube informers and add the pertinent informers
+		switch t := requiredObj.(type) {
+		case *corev1.Namespace:
+			ret = ret.AddNamespaceInformer(informer.Core().V1().Namespaces().Informer(), t.Name)
+		case *corev1.Service:
+			ret = ret.AddInformer(informer.Core().V1().Namespaces().Informer())
+		case *corev1.Pod:
+			ret = ret.AddInformer(informer.Core().V1().Pods().Informer())
+		case *corev1.ServiceAccount:
+			ret = ret.AddInformer(informer.Core().V1().ServiceAccounts().Informer())
+		case *corev1.ConfigMap:
+			ret = ret.AddInformer(informer.Core().V1().ConfigMaps().Informer())
+		case *corev1.Secret:
+			ret = ret.AddInformer(informer.Core().V1().Secrets().Informer())
+		case *rbacv1.ClusterRole:
+			ret = ret.AddInformer(informer.Rbac().V1().ClusterRoles().Informer())
+		case *rbacv1.ClusterRoleBinding:
+			ret = ret.AddInformer(informer.Rbac().V1().ClusterRoleBindings().Informer())
+		case *rbacv1.Role:
+			ret = ret.AddInformer(informer.Rbac().V1().Roles().Informer())
+		case *rbacv1.RoleBinding:
+			ret = ret.AddInformer(informer.Rbac().V1().RoleBindings().Informer())
+		default:
+			klog.V(4).Infof("unhandled type %T", requiredObj)
```

Comment:
fatal?

Reply:
panic!

Reply:
(panic because when this happen to you, you want to know where exactly you should look (backtrace))

Reply:
After slack discusision it sounds fine. Please add a comment that we will fail in a different way in this case.